test/functional.vader

```diff
@@ -0,0 +1,37 @@
+Execute (plug#shellescape() works without optional arguments):
+  if has('unix')
+    AssertEqual "''", plug#shellescape("")
+    AssertEqual "'foo'\\'''", plug#shellescape("foo'")
+  endif
+
+Execute (plug#shellescape() ignores invalid optional argument):
+  if has('unix')
+    AssertEqual "''", plug#shellescape("", '')
+    AssertEqual "'foo'\\'''", plug#shellescape("foo'", [])
+  endif
+
+Execute (plug#shellescape() depends on the shell):
+  AssertEqual "'foo'\\'''", plug#shellescape("foo'", {'shell': 'sh'})
+  AssertEqual '^"foo''^"', plug#shellescape("foo'", {'shell': 'cmd.exe'})
+  AssertEqual "'foo'''", plug#shellescape("foo'", {'shell': 'powershell.exe'})
+  AssertEqual "'foo'''", plug#shellescape("foo'", {'shell': 'pwsh'})
+
+Execute (plug#shellescape() supports non-trivial cmd.exe escaping):
+  " batchfile
+  AssertEqual '^"^^%%PATH^^%%^"', plug#shellescape("^%PATH^%", {
+  \ 'shell': 'cmd.exe',
+  \ })
+  AssertEqual '^"^^%%PATH^^%%^"', plug#shellescape("^%PATH^%", {
+  \ 'shell': 'cmd.exe',
+  \ 'script': 1,
+  \ })
+  " command prompt
+  AssertEqual '^"^^^%PATH^^^%^"', plug#shellescape("^%PATH^%", {
```

Comment:
Is there enough test cases for `plug#shellescape`?

Reply:
I think it's enough. Thanks. Is there anything else you want to address here?

Reply:
When you merge this PR, note the shell settings for powershell. Neovim has `:h shell-powershell` which uses the same shell settings from my vimrc.

Reply:
I see. It looks like we need to create a Wiki page for that.